sidecar/main.go

```diff
@@ -43,19 +44,21 @@ func main() {
 		csiAddonsAddress   = flag.String("csi-addons-address", "/run/csi-addons/socket", "CSI Addons endopoint")
 		nodeID             = flag.String("node-id", "", "NodeID")
 		stagingPath        = flag.String("stagingpath", defaultStagingPath, "stagingpath")
-		controllerPort     = flag.String("controller-port", "",
+		controllerPort     = flag.Int("controller-port", 0,
 			"The TCP network port where the gRPC server for controller request, will listen (example: `8080`)")
 		controllerIP = flag.String("controller-ip", "",
 			"The TCP network ip address where the gRPC server for controller request, will listen (example: `192.168.61.228`)")
 		podName      = flag.String("pod", "", "name of the Pod that contains this sidecar")
 		podNamespace = flag.String("namespace", "", "namespace of the Pod that contains this sidecar")
 		podUID       = flag.String("pod-uid", "", "UID of the Pod that contains this sidecar")
+		proxyPort    = flag.Int("proxy-port", 0, "The TCP port that is available for outside connections to the gRPC server (example: `9070`)")
```

Comment:
Initial testing used kube-rbac-proxy that proxied the connection between the sidecar and the controller. As your much enhanced PR does not use kube-rbac-proxy, the `proxy-port` parameter is not needed anymore.